crates/shared/src/price_estimation/trade_verifier.rs

```diff
@@ -477,14 +634,27 @@ impl SettleOutput {
         let tokens = function.decode_output(output).context("decode")?;
         let (gas_used, balances): (U256, Vec<U256>) = Tokenize::from_token(Token::Tuple(tokens))?;
 
-        let settlement_sell_balance_before = u256_to_big_rational(&balances[0]);
-        let settlement_buy_balance_before = u256_to_big_rational(&balances[1]);
-
-        let trader_balance_before = balances[2];
-        let trader_balance_after = balances[3];
+        let mut i = 0;
```

Comment:
I think this is pretty hard to understand without more context.
Worth a comment explaining that balances get stored in following order:
[...tokens_before, user_balance_before, user_balance_after, ...tokens_after]